File: code_oop/car.py
class Car:
    
    # class variable
    wheels = 0
    
    def __init__(self, color, model, year):
        # In Python, you both declare and define attributes inside the class __init__(), 
        # which is the equivalent of Java’s constructor:
        self.color = color
        self.model = model
        self.year = year
        # non-public
        self._voltage = 12      
        
    @property
    def voltage(self):
        return self._voltage

    @voltage.setter
    def voltage(self, volts):
        print("Warning: this can cause problems!")
        self._voltage = volts

    @voltage.deleter
    def voltage(self):
        print("Warning: the radio will stop working!")
        del self._voltage
        
    def __str__(self):
        return f'Car {self.color} : {self.model} : {self.year}'

File: code_oop/Car.java
// Car.java
public class Car {
    /*
     * In Java, you declare attributes in the class body, outside of any methods,
     * with a definite type. You must define class attributes before they are used.
     */
    private String color;
    private String model;
    private int year;
    private static int wheels;

    public Car(String color, String model, int year) {
        this.color = color;
        this.model = model;
        this.year = year;
    }

    public String getColor() {
        return this.color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public String getModel() {
        return this.model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public int getYear() {
        return this.year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public static int getWheels() {
        return wheels;
    }

    public static void setWheels(int count) {
        wheels = count;
    }

    @Override
    public String toString() {
        return "{" +
                " color='" + getColor() + "'" +
                ", model='" + getModel() + "'" +
                ", year='" + getYear() + "'" +
                "}";
    }
}